src/ngraph/codegen/toolchains/Cuda.h

```diff
@@ -0,0 +1,203 @@
+//===--- Cuda.h - Cuda ToolChain Implementations ----------------*- C++ -*-===//
```

Comment:
Is this the stock Cuda.h file? Can't we get it from the normal cuda install header location? Is this from LLVM headers? Can we use those?